scripts/west_commands/run_common.py

```diff
@@ -224,35 +235,46 @@ def do_run_common(command, user_args, user_runner_args, domain_file=None):
-                board_yaml_file = Path(cache['BOARD_DIR']) / 'board.yml'
-                group_type = 'boards'
+                highest_priority = IGNORED_RUN_ONCE_PRIORITY
+                highest_entry = None
+                check_files = []
 
-                # Search for flash runner configuration, board takes priority over SoC
-                try:
-                    with open(board_yaml_file, 'r') as f:
-                        data_yaml = yaml.safe_load(f.read())
+                for directory in re.split(';', cache['SOC_DIRECTORIES']):
+                    check_files.append(SocBoardFilesProcessing(Path(directory) / 'soc.yml'))
+                for directory in re.split(';', cache['BOARD_DIRECTORIES']):
+                    check_files.append(SocBoardFilesProcessing(Path(directory) / 'board.yml', True))
 
-                except FileNotFoundError:
-                    continue
-
-                if 'runners' not in data_yaml:
-                    # Check SoC file
-                    group_type = 'qualifiers'
+                for check in check_files:
                     try:
-                        with open(soc_yaml_file, 'r') as f:
-                            data_yaml = yaml.safe_load(f.read())
+                        with open(check.filename, 'r') as f:
+                            check.yaml = yaml.safe_load(f.read())
+
+                            if 'runners' not in check.yaml:
+                                continue
+                            elif check.board is False and 'run_once' not in check.yaml['runners']:
+                                continue
+
+                            if 'priority' in check.yaml['runners']:
+                                check.priority = check.yaml['runners']['priority']
+                            else:
+                                check.priority = BOARD_FILE_RUN_ONCE_DEFAULT_PRIORITY if check.board is True else SOC_FILE_RUN_ONCE_DEFAULT_PRIORITY
+
+                            if check.priority == highest_priority:
+                                log.die("Duplicate flash run once configuration found with equal priorities")
+
+                            elif check.priority > highest_priority:
+                                highest_priority = check.priority
+                                highest_entry = check
 
                     except FileNotFoundError:
                         continue
 
-                processed_boards.add(cache['BOARD_DIR'])
+                group_type = 'boards' if highest_entry is not None and highest_entry.board is True else 'qualifiers'
 
-                if 'runners' not in data_yaml or 'run_once' not in data_yaml['runners']:
-                    continue
+                processed_boards.add(cache['BOARD_DIR'])
 
-                for cmd in data_yaml['runners']['run_once']:
-                    for data in data_yaml['runners']['run_once'][cmd]:
+                for cmd in highest_entry.yaml['runners']['run_once']:
```

Comment:
`highest_entry` can still be `None` here.

Reply:
Fixed